### PR TITLE
[WIP] make sure query and database files exist

### PR DESCRIPTION
**Pull request by ctSkennerton**
The databases in the lca commands are loaded first which can
take a bit of time (~10s in my use case) and means that there is a delay in
finding out if one of the query files has been incorrectly typed.

This small function checks if all of the database and query files
exist before trying to load anything and saves a small amount of time.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
hey @ctSkennerton thanks for the PR! it looks like the tests are failing because you're using `os.path.isfile` instead of `os.path.exists`, and in at least some cases directories can be passed in as valid lca paths (when `--traverse-directory` is specified), and that fails the `os.path.isfile` test.

**Comment by ctb** [comment]
See https://github.com/ctSkennerton/sourmash/pull/1 for fixes (and update to latest master)!

**Comment by ctSkennerton** [comment]
I didn't even know that I could give directories, that will teach me not to run the test suite. Thanks for fixing it up!

**Comment by ctb** [comment]
:)

File: sourmash/lca/command_gather.py [hunk 1]
```diff
@@ -108,7 +107,7 @@ def gather_signature(query_sig, dblist, ignore_abundance):
                     md5 = lca_db.lineage_id_to_signature[lid]
                     signature_size = lca_db.lineage_id_counts[lid]
                     assignments[hashval].add((md5, signature_size))
-                    
+
```
Comment (ctb) [inline sourmash/lca/command_gather.py]: unnecessary space :)

File: sourmash/lca/command_gather.py [hunk 2]
```diff
@@ -172,7 +171,7 @@ def gather_signature(query_sig, dblist, ignore_abundance):
         yield result, f_unassigned, est_bp, query_mins
 
     ## done.
-    
+
```
Comment (ctb) [inline sourmash/lca/command_gather.py]: more space!

**Comment by ctb** [comment]
Welcome! If you could clean up the added spaces (maybe it was my fault... if so sorry) that would be great; but other than that it's ready to merge!

**Comment by ctb** [comment]
hey @ctSkennerton shall I review (and, if ready, merge) this? Or are you planning more?

**Comment by ctSkennerton** [comment]
Please review and tell me if there are things I should fix.

**Pull request merged**